### DISCUSSION TITLE
AOSP 12.0 v400.d

### DISCUSSION THREAD

**Discussion by phhusson** [phhusson/treble_experimentations#2096] [General]
- December SPL
- (from AOSP) fix graphical glitches
- Add alternative fix to disable audio effects that doesn't disable all effects
- OnePlus8 overlay
- New secure variant, with deleted su, and vendor fingerprint

<hr /><em>This discussion was created from the release <a href='https://github.com/phhusson/treble_experimentations/releases/tag/v400.d'>AOSP 12.0 v400.d</a>.</em>

**Comment by rvlphee** [comment]
what does "secure" mean?

**Reply by devrieshh** [reply]
secure,where you can use banking apps ed

**Reply by Aarqw12** [reply]
> what does "secure" mean?

Secure variant mean their is no root ( su ) enabled in gsi ( more securized ) . Its good for pass safetynet and app require no root for run.

**Reply by bukyapavankumar** [reply]
cheers @phhusson everything works on redmi note 10 pro . it is laggy .but it is  expected from a prerelease build.

**Comment by ghost** [comment]
Good for daily use?

**Reply by Marcos8760** [reply]
Not yet. I would wait for a full release before you daily drive this.

**Reply by Vinny24** [reply]
May i port this ROM to Samsung A10s which is MT6762 Helio P22?

**Comment by tnolte1** [comment]
Thanks! The secure editions are a great thing. I installed the arm64-ab-vndklite-gapps-secure on my Blackview A80Plus and now the device is Play Certified and GPay and other banking apps work!

**Comment by Tanbohu8** [comment]
I need helps:

C:\Program Files (x86)\Minimal ADB and Fastboot>fastboot flash system "C:\Program Files (x86)\Minimal ADB and Fastboot\dotOS-R-v5.2-arm64-ab-GAPPS-OFFICIAL.img.xz"
target reported max download size of 134217728 bytes
Invalid sparse file format at header magic
error: write_sparse_skip_chunk: don't care size 972804376 is not a multiple of the block size 4096
error: write_sparse_skip_chunk: don't care size 838590744 is not a multiple of the block size 4096
error: write_sparse_skip_chunk: don't care size 704377112 is not a multiple of the block size 4096
error: write_sparse_skip_chunk: don't care size 570163480 is not a multiple of the block size 4096
error: write_sparse_skip_chunk: don't care size 435949848 is not a multiple of the block size 4096
error: write_sparse_skip_chunk: don't care size 301736216 is not a multiple of the block size 4096
error: write_sparse_skip_chunk: don't care size 167522584 is not a multiple of the block size 4096
error: write_sparse_skip_chunk: don't care size 33308952 is not a multiple of the block size 4096
erasing 'system'...
OKAY [  0.118s]
sending sparse 'system' 1/9 (131068 KB)...
error: write_sparse_skip_chunk: don't care size 972804376 is not a multiple of the block size 4096
error: write_sparse_skip_chunk: don't care size 972804376 is not a multiple of the block size 4096
OKAY [  5.727s]
writing 'system' 1/9...
FAILED (remote: sparse chunk size is too big.)
finished. total time: 7.621s

**Reply by rlucianord** [reply]
Hi, I had the same problem, I was doing bad, you have to get your device adb reboot fastboot, it works for me.

**Comment by devrieshh** [comment]
Did you un xz the file...

**Comment by Keviny9** [comment]
You will make the GSI Upgradable with System Update?

**Comment by redymedan** [comment]
Hi, how to fix device isn't play protect certified? help please

**Reply by bukyapavankumar** [reply]
Use secure version

**Reply by redymedan** [reply]
thanks for your reply, how about other variants, I want to install PE based on AOSP but the problem is same device isn't play protect certified

**Reply by tosuman** [reply]
what version is secure version I've also the issue device is not certified

**Reply by Marcos8760** [reply]
> what version is secure version I've also the issue device is not certified

the secure version doesn't have su and fixes the play protect issue.

**Reply by tosuman** [reply]
and also the secure version is not available for ab vndk it is only for vndklite

**Reply by tosuman** [reply]
I'm using PE but I've the only issue with my phone is JIO sim(volte) issue can't call and message also

**Comment by devrieshh** [comment]
You can try to register your phone by Google,you have to find out how

**Comment by phhusson** [comment]
Register your device on https://www.google.com/android/uncertified/

To get the ID, you can either:
- adb root, then adb shell, press enter, and at the new prompt, inside adb
shell, do sqlite3 /data/data/com.google.android.gsf/databases/gservices.db
   'select * from main where name = "android_id";'
- for multiuser see
https://github.com/phhusson/treble_experimentations/issues/1794#issuecomment-816632811

**Comment by tosuman** [comment]
please fix the volte issue in exynos processor like 9609/9610 I can't use GSI for the problem As well as my phone dont have so much custom rom support. Please fix the issue

**Reply by tosuman** [reply]
my Phone is Motorola One Action Not working Jio(Volte) please fix the issue if someone can

**Reply by bukyapavankumar** [reply]
Did you try to install IMS.apk from settings?

**Reply by tosuman** [reply]
yes, but it's not installing IMS it's crashing every time

**Reply by tosuman** [reply]
I've the same issue with all GSI have on github with android 11 & 12

**Reply by bukyapavankumar** [reply]
No. First time it ask for permission to install and crashes .then it installls fine next time I try

**Reply by tosuman** [reply]
sir but I've been trying so much but each an every time it crashes

**Reply by bukyapavankumar** [reply]
Idk then .maybe a device specific issue.

**Comment by Mustafa38000** [comment]
Hi i was wondering which android 12 version is this aosp based on ? And which version was v400.c based on?

**Reply by phhusson** [reply]
12.0.0_r16

**Comment by crazo7924** [comment]
Installed as a DSU. Works alright but MTK IMS with R vendor VoLTE fails to get registered. Tried installing your ims under settings but no success.

**Comment by rlucianord** [comment]
Hi, I'm testing this GSI, get in mind you have to download the secure ones if not, Play services will fail, another thing I want to know is how Fix the bluetooth problem, it has no sound.

**Reply by devrieshh** [reply]
dont you just have to register your phone by google?
 https://www.google.com/android/uncertified/
